### PR TITLE
Subscription feature tests

### PR DESCRIPTION
**Pull request by miasma13**
Task/Issue URL: https://app.asana.com/0/1200019156869587/1207743313459997/f

**Description**:
Client side iOS Subscription feature tests.

**Steps to test this PR**:
1. Ensure normal app target builds correctly
2. Run full Unit tests suite.

**Definition of Done (Internal Only)**:

* [x] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGoTests/Subscription/StorePurchaseManagerTests.swift [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        storePurchaseManager = DefaultStorePurchaseManager()
+    }
+
+    override func tearDownWithError() throws {
+        storePurchaseManager = nil
+        session = nil
+    }
+
+    func testSubscriptionOptionsWhenNoCachedProducts() async throws {
+        let subscriptionOptions = await storePurchaseManager.subscriptionOptions()
```
Comment (aataraxiaa) [inline DuckDuckGoTests/Subscription/StorePurchaseManagerTests.swift:54]: Minor: Do we not annotate tests on iOS with `GIVEN-WHEN-THEN`?

Comment (aataraxiaa) [reply DuckDuckGoTests/Subscription/StorePurchaseManagerTests.swift:54]: Actually, I see. you annotate `SubscriptionPagesUseSubscriptionFeatureTests`, so maybe just an oversight here.

File: DuckDuckGoTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift [hunk 2]
```diff
@@ -21,26 +21,919 @@ import XCTest
...
+        subscriptionService.confirmPurchaseResult = .success(ConfirmPurchaseResponse(email: Constants.email,
+                                                                                     entitlements: Constants.entitlements,
+                                                                                     subscription: SubscriptionMockFactory.subscription))
+
+        // When
+        let subscriptionSelectedParams = ["id": "some-subscription-id"]
+        let result = await feature.subscriptionSelected(params: subscriptionSelectedParams, original: Constants.mockScriptMessage)
+
+        // Then
+        // TODO: Check pixel fired: DailyPixel.fireDailyAndCount(pixel: .privacyProPurchaseAttempt)
```
Comment (aataraxiaa) [inline DuckDuckGoTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift]: Assuming these TODOs are just temporary while you review pixels.

**Review by aataraxiaa — COMMENTED** [review]
All looks good at a high-level @miasma13 , no red flags. I’ll review more in-depth tomorrow. Left one minor comment.

File: DuckDuckGoTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift (hunk 2 above)
Comment (miasma13) [reply DuckDuckGoTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift]: Correct, they will be gone as I do the pass for checking pixels.

File: DuckDuckGoTests/Subscription/StorePurchaseManagerTests.swift (hunk 1 above)
Comment (miasma13) [reply DuckDuckGoTests/Subscription/StorePurchaseManagerTests.swift:54]: Will update that.

**Review by aataraxiaa — APPROVED** [review]
Looks good @miasma13 👍🏼. Needs a main rebase/merge and then should be good to go.

**Pull request merged**